x-pack/plugins/security_solution/public/timelines/components/timeline/body/unified_timeline_body.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ComponentProps, ReactElement } from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
+import { RootDragDropProvider } from '@kbn/dom-drag-drop';
+import { isEmpty } from 'lodash';
+import { StyledTableFlexGroup, StyledTableFlexItem } from '../unified_components/styles';
+import { UnifiedTimelineComponent } from '../unified_components';
+import { defaultUdtHeaders } from '../unified_components/default_headers';
+import type { PaginationInputPaginated, TimelineItem } from '../../../../../common/search_strategy';
+
+interface UnifiedTimelineBodyProps extends ComponentProps<typeof UnifiedTimelineComponent> {
+  header: ReactElement;
+  pageInfo: Pick<PaginationInputPaginated, 'activePage' | 'querySize'>;
+}
+
+export const UnifiedTimelineBody = (props: UnifiedTimelineBodyProps) => {
+  const {
+    header,
+    pageInfo,
+    columns,
+    rowRenderers,
+    timelineId,
+    itemsPerPage,
+    itemsPerPageOptions,
+    sort,
+    events,
+    refetch,
+    dataLoadingState,
+    totalCount,
+    onEventClosed,
+    expandedDetail,
+    showExpandedDetails,
+    onChangePage,
+    activeTab,
+    updatedAt,
+  } = props;
+
+  const [pageRows, setPageRows] = useState<TimelineItem[][]>([]);
+
+  const rows = useMemo(() => pageRows.flat(), [pageRows]);
+
+  useEffect(() => {
```

Comment:
Not necessary now, but a lot of this logic should live in redux

Reply:
thanks. I will leave it for now as it is.